File: Leetcode Daily Challenge/024-June-2025/014-maximum-difference-by-remapping-a-digit.cpp
// https://leetcode.com/problems/maximum-difference-by-remapping-a-digit/




#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    // TC : O(log(num))
    // SC : O(log(num))
    int minMaxDifference(int num) {
        string a = to_string(num);
        string b = to_string(num);

        int idx = a.find_first_not_of('9');
        if (idx != string::npos) {
            char ch = a[idx];
            replace(begin(a), end(a), ch, '9');
        }

        char ch = b[0];
        replace(begin(b), end(b), ch, '0');

        return stoi(a) - stoi(b); 
    }
};

File: Leetcode Daily Challenge/024-June-2025/014-maximum-difference-by-remapping-a-digit.java
// https://leetcode.com/problems/maximum-difference-by-remapping-a-digit/




class Solution {
    // TC : O(log(num))
    // SC : O(log(num))
    public int minMaxDifference(int num) {
        String a = Integer.toString(num);
        String b = Integer.toString(num);

        int idx = -1;
        for (int i = 0; i < a.length(); i ++) {
            if (a.charAt(i) != '9') {
                idx = i;
                break;
            }
        }
        if (idx != -1) {
            char ch = a.charAt(idx);
            a = a.replace(ch, '9');
        }

        char chr = b.charAt(0);
        b = b.replace(chr, '0');

        return Integer.parseInt(a) - Integer.parseInt(b);
    }
}